Model/ResourceModel/NodeType/Product.php

```diff
@@ -0,0 +1,132 @@
+<?php
+
+namespace Snowdog\Menu\Model\ResourceModel\NodeType;
+
+use Magento\Framework\DB\Adapter\AdapterInterface;
+use Magento\Store\Model\Store;
+use Magento\Framework\App\ResourceConnection;
+use Magento\Catalog\Api\Data\ProductInterface;
+use Magento\Catalog\Model\Product as CoreProduct;
+use Magento\Framework\EntityManager\MetadataPool;
+
+class Product extends AbstractNode
+{
+    /**
+     * @var MetadataPool
+     */
+    private $metadataPool;
+
+    public function __construct(
+        ResourceConnection $resource,
+        MetadataPool $metadataPool
+    ) {
+        $this->metadataPool = $metadataPool;
+        parent::__construct($resource);
+    }
+
+    /**
+     * @return array
+     * @throws \Exception
+     */
+    public function fetchConfigData()
+    {
+        $metadata = $this->metadataPool->getMetadata(ProductInterface::class);
+        $identifierField = $metadata->getIdentifierField();
+        $connection = $this->getConnection('read');
+
+        $select = $connection->select()->from(
```

Comment:
This query can break the store if the catalog has a lot of products.

Reply:
My idea is add grid instead of search by name/sku/id or remove option autocomplete what You think ?

Reply:
Removed